mmengine/model/base_model/base_model.py

```diff
@@ -177,23 +177,17 @@ def parse_losses(
 
         return loss, log_vars  # type: ignore
 
-    def to(self,
-           device: Optional[Union[int, str, torch.device]] = None,
-           *args,
-           **kwargs) -> nn.Module:
+    def to(self, *args, **kwargs) -> nn.Module:
         """Overrides this method to call :meth:`BaseDataPreprocessor.to`
         additionally.
 
-        Args:
-            device (int, str or torch.device, optional): the desired device
-                of the parameters and buffers in this module.
-
         Returns:
             nn.Module: The model itself.
         """
+        device = torch._C._nn._parse_to(*args, **kwargs)[0]
```

Comment:
Using an internal function may cause a breaking issue.

Reply:
I've tried writing a parsing method myself, but finally I believe this internal function is more stable. [Lightning](https://github.com/Lightning-AI/lightning/blob/master/src/lightning_lite/utilities/device_dtype_mixin.py#L52) also uses it directly.

Reply:
What is the minimum pytorch version that provides this interface, and is it greater than or equal to 1.6?

Reply:
I checked pytorch repo with their tags. It appears since v1.0.0